app/temporal/src/domain/DataAnalyzer.ts

```diff
@@ -52,6 +57,75 @@ export class DataAnalyzer {
     return [...foundColumnRecommendations, ...notFoundMatchRecommendations];
   }
 
+  public processDataValidations(
+    data: Record<string, unknown>[],
+    validatorColumns: ValidatorColumns,
+    stats: Stats
+  ): { rowId: number; column: string; errors: ValidationError[] }[] {
+    const chunkErrors: {
+      rowId: number;
+      column: string;
+      errors: ValidationError[];
+    }[] = [];
+    for (const row of data) {
+      for (const validatorKey of Object.keys(
```

Comment:
Object.keys würde ich vor die obere for loop ziehen, ich glaube der Aufruf ist nicht billig (bei 150k)